orchestra/utils/task_lifecycle.py

```diff
@@ -275,6 +228,182 @@ def assign_task(worker_id, task_id):
     return task
 
 
+@transaction.atomic
+def reassign_assignment(worker_id, assignment_id):
+    worker = Worker.objects.get(id=worker_id)
+    assignment = TaskAssignment.objects.get(id=assignment_id)
+    if assignment.assignment_counter > 0:
+        role = WorkerCertification.Role.REVIEWER
+    else:
+        role = WorkerCertification.Role.ENTRY_LEVEL
+
+    if not _worker_certified_for_task(worker, assignment.task, role):
+        raise WorkerCertificationError('Worker not certified for this task.')
+    if is_worker_assigned_to_task(worker, assignment.task):
+        raise TaskAssignmentError('Worker already assigned to this task.')
+
+    assignment.worker = worker
+    assignment.save()
+    add_worker_to_project_team(worker, assignment.task.project)
+
+    return assignment
+
+
+@transaction.atomic
+def complete_and_skip_task(task_id):
+    task = Task.objects.get(id=task_id)
+    task.status = Task.Status.COMPLETE
+    task.save()
+    for assignment in task.assignments.all():
+        assignment.status = TaskAssignment.Status.SUBMITTED
+        assignment.save()
+    create_subsequent_tasks(task.project)
+    return task
+
+
+@transaction.atomic
+def revert_task_to_datetime(task_id, revert_datetime, fake=False):
+    task = Task.objects.get(id=task_id)
+    audit = {'task': task, 'change': None, 'assignments': []}
+
+    assignments_after = (assignment_history(task)
+                         .filter(start_datetime__gt=revert_datetime)
+                         .all())
+
+    if assignments_after.exists():
+        for assignment in assignments_after:
+            audit['assignments'].append({'assignment': assignment,
+                                         'change': 'deleted',
+                                         'snapshots': []})
+        if not fake:
+            # Delete assignments created after the revert datetime.
+            assignments_after.delete()
```

Comment:
any time i see a delete, i get scared because of cascading deletes.  can you assure this doesn't cascade to other objects/models?  happy to explain tricks for doing this, but one simple one is to delete an assignment object in the admin and see whether it warns you about cascading deletes